scripts/zones/Temenos/npcs/Armoury_Crate.lua

```diff
@@ -815,241 +1179,279 @@ local loot =
+                                if j ~= 1 then GetNPCByID(ID.npc.TEMENOS_N_CRATE[i]+1):setStatus(tpz.status.DISAPPEAR) end
+                                if j ~= 2 then GetNPCByID(ID.npc.TEMENOS_N_CRATE[i]+2):setStatus(tpz.status.DISAPPEAR) end
+                                if model == 960 then
+                                    tpz.battlefield.HealPlayers(battlefield)
+                                elseif model == 961 then
+                                    tpz.limbus.handleLootRolls(battlefield, loot[bfid][i], nil, npc)
+                                elseif model == 962 then
+                                    tpz.limbus.extendTimeLimit(battlefield, 15, tpz.zone.TEMENOS)
+                                end
+                            end
+                        end
+                    end
+                else
+                    tpz.limbus.handleLootRolls(battlefield, loot[bfid][7], nil, npc)
+                    battlefield:setLocalVar("cutsceneTimer", 10)
+                    battlefield:setLocalVar("lootSeen", 1)
+                end
+            end,
+            [1300] = function() -- Temenos East Crate Handling
+                local spawnMimic = math.random(0,1) == 1
+                if crateID ~= ID.npc.TEMENOS_E_CRATE[7] and crateID ~= ID.npc.TEMENOS_E_CRATE[7]+1 then
+                    for i = 1, 6 do
+                        local mask = battlefield:getLocalVar("crateMaskF"..i)
+                        for j = 0, 3 do
+                            if crateID == ID.npc.TEMENOS_E_CRATE[i]+j then
+                                if GetMobByID(ID.mob.TEMENOS_E_MOB[i]+4):isDead() then
+                                    battlefield:setLocalVar("crateOpenedF"..i, 1)
+                                    if model ~= 961 or not (mask > 7 and spawnMimic) then
+                                        if j ~= 0 then GetNPCByID(ID.npc.TEMENOS_E_CRATE[i]):setStatus(tpz.status.DISAPPEAR) end
+                                        if j ~= 1 then GetNPCByID(ID.npc.TEMENOS_E_CRATE[i]+1):setStatus(tpz.status.DISAPPEAR) end
+                                        if j ~= 2 then GetNPCByID(ID.npc.TEMENOS_E_CRATE[i]+2):setStatus(tpz.status.DISAPPEAR) end
+                                        if j ~= 3 then GetNPCByID(ID.npc.TEMENOS_E_CRATE[i]+3):setStatus(tpz.status.DISAPPEAR) end
+                                    end
+                                    if GetMobByID(ID.mob.TEMENOS_E_MOB[i]):isAlive() then DespawnMob(ID.mob.TEMENOS_E_MOB[i]) end
+                                    if GetMobByID(ID.mob.TEMENOS_E_MOB[i]+1):isAlive() then DespawnMob(ID.mob.TEMENOS_E_MOB[i]+1) end
+                                    if GetMobByID(ID.mob.TEMENOS_E_MOB[i]+2):isAlive() then DespawnMob(ID.mob.TEMENOS_E_MOB[i]+2) end
+                                    if GetMobByID(ID.mob.TEMENOS_E_MOB[i]+3):isAlive() then DespawnMob(ID.mob.TEMENOS_E_MOB[i]+3) end
```

Comment:
Me: "Hrm, fake chests - this PR is starting to get a little hard to review."
Me: "...Wait, how are we even handling models? They're hard set based on the NPC ID... right?"
_checks_
Me: "Okay, we can do this. We can have random chest models, mimics, pretty code, and without looping. Just... think... and get that review in."
_starts going through Limbus global_
Me: "Nope. Can't get this thoroughly reviewed by Thursday. Punting this to a branch, and I'll open my own PR for logic simplifications. Not gonna make cocosolos put up with my usual nonsense during review of this PR!"

Reply:
I had this stuff in loops originally, but it was too slow and allowed opening multiple chests by timing correctly or spamming. Some floors change the chest model dynamically.